Automattic/wp-calypso
# Congrats: Refactored domain only congrats page

Related to https://github.com/Automattic/dotcom-forge/issues/5119

## Proposed Changes

This PR refactors the Domain-Only Congrats Page to adapt the new approach for unifying the congratulation pages. See pb5gDS-38y-p2#comment-4475 for details.

* Created a `<ThankYouV2>` component that defines a general structure for the Congrats Page. Much of the style was taken from `<ThankYouLayout>`. Ultimately, the idea is to remove `<ThankYouLayout>`.
* Introduced `isRefactored()` util function allowing us to gradually refactor existing pages to use the new framework for rendering Congrats pages.
* Re-created the `<DomainOnlyThankYou>` component that utilizes `<ThankYouV2>` to build out the Congrats page.
* Re-created the `<ProductDomain>` component that utilizes a generic `<ThankYouProduct>` component to build out a domain product tile.
* Updated the back button label to "Back to dashboard" (instead of "Back to home") as per p9Jlb4-7Hx-p2#comment-10510
* Updated click tracking events (see below for details)

**Update 2023-01-11:**

Note that the in production, the footer links are tracked with these events, which is page specific and referencing the wrong page (Domain Transfer to User, instead of Domain Only):
https://github.com/Automattic/wp-calypso/blob/f410e61e9c846719c7ae532fd7fb8e367577b833/client/my-sites/checkout/checkout-thank-you/redesign-v2/sections/footer/DomainOnlyFooter.tsx#L19 

https://github.com/Automattic/wp-calypso/blob/f410e61e9c846719c7ae532fd7fb8e367577b833/client/my-sites/checkout/checkout-thank-you/redesign-v2/sections/footer/DomainOnlyFooter.tsx#L30-L32

Since the footer will be shared by different pages, and there isn't much value in the granular data of which exact page the link was clicked on, in this PR, I've updated the click even names to something page agnostic, matching the respective element keys:

https://github.com/Automattic/wp-calypso/blob/5940d4c993908aaf3138030ca7b31fe36c97ec45/client/my-sites/checkout/checkout-thank-you/redesign-v2/pages/content/get-domain-footer-details.ts#L16

https://github.com/Automattic/wp-calypso/blob/5940d4c993908aaf3138030ca7b31fe36c97ec45/client/my-sites/checkout/checkout-thank-you/redesign-v2/pages/content/get-domain-footer-details.ts#L28

There's barely any clicks to those links at the moment (check out tracks), so I don't think there's a need to maintain historical data with the same event names.

## Testing Instructions



* Run the https://github.com/Automattic/wp-calypso/pull/86186#issuecomment-1883657252 or checkout this branch and run it locally.
* Purchase a domain through the onboarding flow `/start/domain/domain-only` or through **Upgrades > Domains**. Do not include any plans. 
* Complete checkout and assert that you see the Congrats Page without errors, and that all buttons and links work as expected (you can compare the behaviour with production).

<img width="1046" alt="Screen Shot 2024-01-09 at 4 00 50 PM" src="https://github.com/Automattic/wp-calypso/assets/104910361/b894f9d5-2fee-4f8b-8f05-fba2e8759619">
